### PR TITLE
Verify signature with compressed public key

### PR DESCRIPTION
**Pull request by gumb0**
We'll need to verify ENR signature having only compressed public key (33 bytes) and `[r, s]` part of the signature (64 bytes)

For reference go-ethereum code doing the same: https://github.com/ethereum/go-ethereum/blob/c4109d790ffd26d67feb7745d4af8a8bc5090bd9/crypto/secp256k1/ext.h#L46-L63

### CODE REVIEW

File: libdevcrypto/Common.h [hunk 1]
```diff
@@ -133,6 +133,9 @@ Signature sign(Secret const& _k, h256 const& _hash);
 /// Verify signature.
 bool verify(Public const& _k, Signature const& _s, h256 const& _hash);
 
+// Verify signature with compressed public key
+bool verify(PublicCompressed const& _key, h512 const& _signature, h256 const& _hash);
```
Comment (halfalicious) [inline libdevcrypto/Common.h]: Should we name the args similar to the other verify function i.e. _k/_s?

File: test/unittests/libdevcrypto/crypto.cpp [hunk 2]
```diff
@@ -181,6 +181,34 @@ BOOST_AUTO_TEST_CASE(verify_secert)
 	BOOST_REQUIRE(k.address());
 }
 
+BOOST_AUTO_TEST_CASE(verifyWithPublicCompressed)
+{
+    PublicCompressed pub{
```
Comment (halfalicious) [inline test/unittests/libdevcrypto/crypto.cpp]: Presumably you encrypted the hash into the hex value you're using as the signature value outside of the test case...can we include the encryption as part of the test?

File: test/unittests/libdevcrypto/crypto.cpp [hunk 3]
```diff
@@ -181,6 +181,34 @@ BOOST_AUTO_TEST_CASE(verify_secert)
...
+    BOOST_REQUIRE(verify(pub, sig, msg));
+
+    h512 invalidSig{sig};
+    ++invalidSig[4];
+    BOOST_REQUIRE(!verify(pub, invalidSig, msg));
+}
+
+BOOST_AUTO_TEST_CASE(signAndVerify)
+{
+    KeyPair kp = KeyPair::create();
```
Comment (halfalicious) [inline test/unittests/libdevcrypto/crypto.cpp]: (Nit) can be auto

File: test/unittests/libdevcrypto/crypto.cpp [hunk 4]
```diff
@@ -181,6 +181,34 @@ BOOST_AUTO_TEST_CASE(verify_secert)
...
+    auto msg = h256::random();
```
Comment (halfalicious) [inline test/unittests/libdevcrypto/crypto.cpp]: (Nit) Can msg and sig be const?

File: test/unittests/libdevcrypto/crypto.cpp [hunk 5]
```diff
@@ -181,6 +181,34 @@ BOOST_AUTO_TEST_CASE(verify_secert)
...
+    auto sig = sign(kp.secret(), msg);
+
+    auto pubCompressed = toPublicCompressed(kp.secret());
```
Comment (halfalicious) [inline test/unittests/libdevcrypto/crypto.cpp]: (Nit) can pubCompressed be const?

File: test/unittests/libdevcrypto/crypto.cpp (hunk 2 above)
Comment (gumb0) [reply test/unittests/libdevcrypto/crypto.cpp]: It's what `signAndVerify` test does

**Pull request merged**